ext/openssl/xp_ssl.c

```diff
@@ -908,15 +908,7 @@ static int php_openssl_set_local_cert(SSL_CTX *ctx, php_stream *stream) /* {{{ *
 			}
 			GET_VER_OPT_STRING("local_pk", private_key);
 
-			if (private_key) {
-				char resolved_path_buff_pk[MAXPATHLEN];
-				if (VCWD_REALPATH(private_key, resolved_path_buff_pk)) {
-					if (SSL_CTX_use_PrivateKey_file(ctx, resolved_path_buff_pk, SSL_FILETYPE_PEM) != 1) {
-						php_error_docref(NULL, E_WARNING, "Unable to set private key file `%s'", resolved_path_buff_pk);
-						return FAILURE;
-					}
-				}
-			} else {
+			if (!private_key || VCWD_REALPATH(private_key, resolved_path_buff)) {
```

Comment:
Is it possible for VCWD_REALPATH to write something to resolved_path_buff even if it fails?

I also don't think it was the intention of this code to simply do nothing if local_pk points to a file that doesn't exist. I think that should throw a warning and return failure.